L1Trigger/L1TTrackMatch/plugins/L1FastTrackingJetProducer.cc

```diff
@@ -0,0 +1,242 @@
+  bool doTightChi2_;
+  bool displaced_;            //use prompt/displaced tracks
+  bool selectTrkMatchGenTight_;
+  bool selectTrkMatchGenLoose_;
+  bool selectTrkMatchGenOrPU_;
+
+  const edm::EDGetTokenT<std::vector<TTTrack< Ref_Phase2TrackerDigi_ > > > trackToken_;
+  edm::EDGetTokenT<std::vector<l1t::Vertex>> pvToken_;
+  const edm::EDGetTokenT<TTTrackAssociationMap<Ref_Phase2TrackerDigi_> > genToken_;
+};
+
+// constructor
+L1FastTrackingJetProducer::L1FastTrackingJetProducer(const edm::ParameterSet& iConfig) :
+trackToken_(consumes< std::vector<TTTrack< Ref_Phase2TrackerDigi_> > > (iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))),
+pvToken_(consumes<std::vector<l1t::Vertex>>(iConfig.getParameter<edm::InputTag>("L1PrimaryVertexTag"))),
+genToken_(consumes< TTTrackAssociationMap<Ref_Phase2TrackerDigi_> > (iConfig.getParameter<edm::InputTag>("GenInfo")))
+{
+  trkZMax_    = (float)iConfig.getParameter<double>("trk_zMax");
+  trkChi2dofMax_ = (float)iConfig.getParameter<double>("trk_chi2dofMax");
+  trkBendChi2Max_ = iConfig.getParameter<double>("trk_bendChi2Max");
+  trkPtMin_   = (float)iConfig.getParameter<double>("trk_ptMin");
+  trkEtaMax_  = (float)iConfig.getParameter<double>("trk_etaMax");
+  trkNStubMin_   = (int)iConfig.getParameter<int>("trk_nStubMin");
+  trkNPSStubMin_ = (int)iConfig.getParameter<int>("trk_nPSStubMin");
+  deltaZ0Cut_ = (float)iConfig.getParameter<double>("deltaZ0Cut");
+  coneSize_ = (float)iConfig.getParameter<double>("coneSize");
+  doTightChi2_ = iConfig.getParameter<bool>("doTightChi2");
+  displaced_ = iConfig.getParameter<bool>("displaced");
+  selectTrkMatchGenTight_ = iConfig.getParameter<bool> ("selectTrkMatchGenTight");
+  selectTrkMatchGenLoose_ = iConfig.getParameter<bool> ("selectTrkMatchGenLoose");
+  selectTrkMatchGenOrPU_ = iConfig.getParameter<bool> ("selectTrkMatchGenOrPU");
+  if (displaced_) produces<TkJetCollection>("L1FastTrackingJetsExtended");
+  else produces<TkJetCollection>("L1FastTrackingJets");
+}
+
+// destructor
+L1FastTrackingJetProducer::~L1FastTrackingJetProducer() { }
+
+// producer
+void L1FastTrackingJetProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  std::unique_ptr<TkJetCollection> L1FastTrackingJets(new TkJetCollection);
+
+  // L1 tracks
+  edm::Handle< std::vector< TTTrack< Ref_Phase2TrackerDigi_ > > > TTTrackHandle;
+  iEvent.getByToken(trackToken_, TTTrackHandle);
+  std::vector< TTTrack< Ref_Phase2TrackerDigi_ > >::const_iterator iterL1Track;
+
+  // Gen
+  edm::Handle< TTTrackAssociationMap<Ref_Phase2TrackerDigi_> > MCTrkAssociation;
+  if (!iEvent.isRealData())
+      iEvent.getByToken(genToken_,MCTrkAssociation);
+
+  // Tracker Topology
+  edm::ESHandle<TrackerTopology> tTopoHandle_;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle_);
+  const TrackerTopology* tTopo = tTopoHandle_.product();
+  ESHandle<TrackerGeometry> tGeomHandle;
+  iSetup.get<TrackerDigiGeometryRecord>().get(tGeomHandle);
```

Comment:
Please migrate to EventSetup consumes

Reply:
Dear Cecile,

Thank you for the suggestion, I will do it. Just to be clear, you mean the tracker topology or the geometry (or both)? Furthermore is somewhere an example to see the correct syntax? Thanks.

Best,
George

Reply:
Dear George, I meant both the tracker topology and geometry. In general it is recommended now to use EDGetToken instead of ESHandle. You can find examples of usage in recent CMSSW code. And for example these are changes I made to satisfy this new constraint: https://github.com/cms-sw/cmssw/pull/33612/commits/29bdf6f1f4ea9e47faa6a7279b8ef26b2c4d3e66

Reply:
Dear Cecile,

Thank you very much for the comment. I applied it as you said and confirmed that the result is identical. Thanks.

Best,
George